code/lib/channels/src/postmessage/getEventSourceUrl.ts

```diff
@@ -7,6 +7,14 @@ export const getEventSourceUrl = (event: MessageEvent) => {
   // try to find the originating iframe by matching it's contentWindow
   // This might not be cross-origin safe
   const [frame, ...remainder] = frames.filter((element) => {
+    try {
+      return (
+        element.contentWindow?.location.origin === (event.source as Window).location.origin &&
+        element.contentWindow?.location.pathname === (event.source as Window).location.pathname
+      );
+    } catch (err) {
+      // continue
+    }
```

Comment:
I added this block because when you have 2 iFrames with the same origin, it seems they might share a [WindowProxy](https://developer.mozilla.org/en-US/docs/Glossary/WindowProxy)

I tested this code manually, the `pathname` comparison does the trick.